### ISSUE TITLE
AND Gate doesn't recalculate redstone state until it receives a block update

### ISSUE THREAD

**Issue by TheDeviantCrafter** [asiekierka/CharsetMC#203]
This may affect other gates as well, but I haven't tested them. If you place an AND gate and supply a redstone signal to all three input sides, it will not give an output signal. All four redstone torches will be off.

As soon as you give it a block update, it will realize that the output signal should be on and update accordingly. It should update when it receives a redstone signal, but it doesn't.

Here's an image of a setup before a block update: https://i.imgur.com/VevsBSj.png

Here's the same setup after I place a block above the AND gate and broke it: https://i.imgur.com/RCtVk8d.png

**Comment by asiekierka** [comment]
EDIT: I'm an idiot. Of course this was a Charset bug.

**Comment by TheDeviantCrafter** [comment]
I repeated the same experiment using redstone dust, and the result was the same. When the redstone dust changes its power level it doesn't update the state of the AND gate.

This is the setup I used: https://imgur.com/a/jGSs9

In that screenshot, it's in an invalid state. It updates correctly once I give it a block update.

Edit: To clarify, flipping the levers didn't change the state of the AND gate.
